data/src/test/java/com/linkedin/data/codec/entitystream/TestProtobufDataDecoder.java

```diff
@@ -0,0 +1,184 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.linkedin.data.codec.entitystream;
+
+import com.linkedin.data.ByteString;
+import com.linkedin.data.ChunkedByteStringWriter;
+import com.linkedin.data.DataComplex;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.TestUtil;
+import com.linkedin.data.codec.CodecDataProviders;
+import com.linkedin.data.codec.ProtobufCodecOptions;
+import com.linkedin.data.codec.ProtobufDataCodec;
+import com.linkedin.entitystream.EntityStream;
+import com.linkedin.entitystream.EntityStreams;
+import com.linkedin.entitystream.Writer;
+import java.util.concurrent.ExecutionException;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+public class TestProtobufDataDecoder
+{
+  @Test(dataProvider = "protobufCodecData", dataProviderClass = CodecDataProviders.class)
+  public void testDecoder(String testName, DataComplex dataComplex, boolean enableFixedLengthFloatDoubles)
+      throws Exception {
+    ProtobufDataCodec codec = new ProtobufDataCodec(
+        new ProtobufCodecOptions.Builder().setEnableFixedLengthFloatDoubles(enableFixedLengthFloatDoubles)
+            .setEnableASCIIOnlyStrings(false).build());
+    byte[] bytes = TestUtil.dataComplexToBytes(codec, dataComplex);
+    DataComplex decodedDataComplex = decode(bytes);
+    assertEquals(TestUtil.dataComplexToBytes(codec, decodedDataComplex), bytes);
+    codec = new ProtobufDataCodec(
+        new ProtobufCodecOptions.Builder().setEnableFixedLengthFloatDoubles(enableFixedLengthFloatDoubles)
+            .setEnableASCIIOnlyStrings(true).build());
```

Comment:
We should take setEnableASCIIOnlyStrings as a dataprovider param

Reply:
actually will remove test cases for setEnableASCIIOnlyStrings=false as this is deprecated, any both ascii and utf-8 string can be tested without this.